Mathlib/CategoryTheory/Limits/FintypeCat.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2023 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.FintypeCat
+import Mathlib.CategoryTheory.Limits.Shapes.FiniteLimits
+import Mathlib.CategoryTheory.Limits.Types
+import Mathlib.CategoryTheory.Limits.Creates
+import Mathlib.CategoryTheory.Limits.Preserves.Finite
+import Mathlib.Data.FunLike.Fintype
+
+/-!
+# (Co)limits in the category of finite types
+
+We show that finite (co)limits exist in `FintypeCat` and that they are preserved by the natural
+inclusion `FintypeCat.incl`.
+-/
+
+open CategoryTheory Limits Functor
+
+universe u
+
+namespace CategoryTheory.Limits.FintypeCat
+
+instance {J : Type} [SmallCategory J] [FinCategory J] (K : J ⥤ FintypeCat.{u}) (j : J) :
+  Finite ((K ⋙ FintypeCat.incl.{u}).obj j) := by
+    simp only [comp_obj, FintypeCat.incl_obj]
+    exact inferInstance
+
+/-- Any functor from a finite category to Types that only involves finite objects,
+has a finite limit. -/
+noncomputable instance finiteLimitOfFiniteDiagram {J : Type} [SmallCategory J] [FinCategory J]
+    (K : J ⥤ Type*) [∀ j, Finite (K.obj j)] : Fintype (limit K) := by
+  have : Fintype (sections K) := Fintype.ofFinite (sections K)
+  exact Fintype.ofEquiv (sections K) (Types.limitEquivSections K).symm
+
+noncomputable instance inclusionCreatesFiniteLimits {J : Type} [SmallCategory J] [FinCategory J] :
+    CreatesLimitsOfShape J FintypeCat.incl.{u} where
+  CreatesLimit := by
+    intro K
+    exact createsLimitOfFullyFaithfulOfIso (FintypeCat.of <| limit <| K ⋙ FintypeCat.incl)
+      (Iso.refl _)
```

Comment:
```suggestion
  CreatesLimit {K} := createsLimitOfFullyFaithfulOfIso
    (FintypeCat.of <| limit <| K ⋙ FintypeCat.incl) (Iso.refl _)
```
This syntax is more compact.

Reply:
Nice, I did not know of the `CreatesLimit {K}` syntax for implicit arguments.